File: src/main/java/ru/ioleksiv/telegram/bot/core/model/telegram/model/type/game/Animation.java
package ru.ioleksiv.telegram.bot.core.model.telegram.model.type.game;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.model.telegram.model.type.files.PhotoSize;

/**
 * @link https://core.telegram.org/bots/api#animation
 */
public class Animation {
    /**
     * file_id	String	Unique file identifier
     */
    @JsonProperty("file_id")
    private String fileId;
    /**
     * thumb	PhotoSize	Optional. Animation thumbnail as defined by sender
     */
    @JsonProperty("thumb")
    private PhotoSize thumb;
    /**
     * file_name	String	Optional. Original animation filename as defined by sender
     */
    @JsonProperty("file_name")
    private String fileName;
    /**
     * mime_type	String	Optional. MIME type of the file as defined by sender
     */
    @JsonProperty("mime_type")
    private String mimeType;
    /**
     * file_size	Integer	Optional. File size
     */
    @JsonProperty("file_size")
    private Integer fileSize;

    public String getFileId() {
        return fileId;
    }

    public void setFileId(String fileId) {
        this.fileId = fileId;
    }

    public PhotoSize getThumb() {
        return thumb;
    }

    public void setThumb(PhotoSize thumb) {
        this.thumb = thumb;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getMimeType() {
        return mimeType;
    }

    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    public Integer getFileSize() {
        return fileSize;
    }

    public void setFileSize(Integer fileSize) {
        this.fileSize = fileSize;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/objects/std/game/Animation.kt
package io.github.oleksivio.tl.kbot.server.api.objects.std.game

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.objects.std.files.PhotoSize
import io.github.oleksivio.tl.kbot.server.api.model.ITelegram

/**
 *  [Animation](https://core.telegram.org/bots/api/#animation)
 */
data class Animation(
    /**
         * file_id String Unique file identifier
         */
        @JsonProperty("file_id")
        var fileId: String? = null,
    /**
         * To setup filter:
         *
         *  PhotoFilter thumb thumb PhotoSize Optional. Animation thumbnail as defined by sender
         */
        @JsonProperty("thumb")
        var thumb: PhotoSize? = null,
    /**
         * To setup filter:
         *
         *  StringFilter fileName file_name [String] Optional. Original animation filename as defined by sender
         */
        @JsonProperty("file_name")
        var fileName: String? = null,
    /**
         * To setup filter:
         *
         *  StringFilter mimeType mime_type [String] Optional. MIME type of the file as defined by sender
         */
        @JsonProperty("mime_type")
        var mimeType: String? = null,
    /**
         * To setup filter:
         *
         *  IntegerFilter fileSize file_size Integer Optional. File size
         */
        @JsonProperty("file_size")
        var fileSize: Int? = null,
    /**
         * To setup filter:
         *
         *  IntegerFilter width width Integer Video width as defined by sender
         */
        @JsonProperty("width")
        var width: Int? = null,
    /**
         * To setup filter:
         *
         *  IntegerFilter height height Integer Video height as defined by sender
         */
        @JsonProperty("height")
        var height: Int? = null,
    /**
         * To setup filter:
         *
         *  IntegerFilter duration duration Integer Duration of the video in seconds as defined by sender
         */
        @JsonProperty("duration")
        var duration: Int? = null
) : ITelegram
